stream/stream_file.c

```diff
@@ -85,6 +86,26 @@ typedef enum _FSINFOCLASS {
 #endif
 #endif
 
+
+struct appending_opts {
+    double retry_timeout;
+    int max_retries;
+};
+
+#define OPT_BASE_STRUCT struct appending_opts
+const struct m_sub_options stream_appending_conf = {
+    .opts = (const struct m_option[]) {
+        {"retry-timeout", OPT_DOUBLE(retry_timeout), M_RANGE(0.1, 1.0)},
```

Comment:
This range seems a bit tight? any reason for that?

Reply:
well because there are multiple retries.

however having both of these options seems needlessly confusing. it would work perfectly fine to have just `retry-timeout` and then automatically the code automatically re-checks every 0.2s

Reply:
It goes against mpv core principles where everything is customization. Though given how small is the range to customize this variable, it makes not much difference in practice, no?

> it would work perfectly fine to have just retry-timeout and then automatically the code automatically re-checks every 0.2s

You meant `max-retries`?

Reply:
> You meant `max-retries`?

well no. then the unit of the option would be 0.2s, which is confusing.
the user should specify which total timeout he wants and then mpv retries as it sees fit (could even be linear backoff).

Reply:
I see.

Reply:
Regarding exposing `retry_timeout` and it's range.

My thought was that a lower value than 0.2 may help allowing uninterrupted playback in cases where a stream is being appended to very often. Think an unsegmented live stream that is being saved as a file, or a live stream with very short segments.

On the upper bound, my thought was allowing for less resource usage if the user knows appending doesn't happen with high frequency, and doesn't mind the delay between retries, which is maybe less convincing, but still a nice to have IMHO.

I think both options should be exposed in any case.

Reply:
I will let @sfan5 decide on this one. For me it is fine to have those options exposed.

Reply:
if the user is interested in micro-optimizing resource usage he should use a pipe, which does not require polling.
I stand by my suggestion to only have one option.